core/src/main/scala/geotrellis/store/query/Repository.scala

```diff
@@ -16,11 +16,24 @@
 
 package geotrellis.store.query
 
-import cats.Id
+import cats.{FlatMap, Functor, Semigroup, SemigroupK}
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.syntax.semigroupk._
+import geotrellis.store.query
 
 trait RepositoryM[M[_], G[_], T] {
-  def store: M[G[T]]
+  def store: M[G[T]] = find(query.all)
   def find(query: Query): M[G[T]]
 }
 
-trait Repository[G[_], T] extends RepositoryM[Id, G, T]
+object RepositoryM {
+  implicit def semigroupRepositoryM[M[_]: FlatMap: Functor, G[_]: SemigroupK, T]
+    : Semigroup[RepositoryM[M, G, T]] =
+    (x: RepositoryM[M, G, T], y: RepositoryM[M, G, T]) =>
+      new RepositoryM[M, G, T] {
+
+        override def find(query: Query): M[G[T]] =
```

Comment:
what do you think about `convention` I'm trying to follow? 
to use `override` only in places where it overrides the default behavior? (like in these `override val store` cases).

Reply:
AH, how do you feel about doing it through [Applicative](https://typelevel.org/cats/typeclasses/applicative.html#applicatives-compose)? What we really need is independent composition, but `Monad` gives us capablities to chain operations.

Reply:
Works for me. Updated. Looks like this was the only spot I did this that I could find.